sandbox/modules/identity/src/main/java/org/opensearch/identity/SecurityRestFilter.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.shiro.authc.AuthenticationException;
+import org.opensearch.OpenSearchException;
+import org.opensearch.authn.Subject;
+import org.opensearch.authn.jwt.JwtVendor;
+import org.opensearch.authn.tokens.AuthenticationToken;
+import org.opensearch.authn.tokens.BasicAuthToken;
+import org.opensearch.authn.tokens.HttpHeaderToken;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SecurityRestFilter {
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+    private final ThreadContext threadContext;
+    private final Settings settings;
+    private final Path configPath;
+
+    public SecurityRestFilter(final ThreadPool threadPool, final Settings settings, final Path configPath) {
+        super();
+        this.threadContext = threadPool.getThreadContext();
+        this.settings = settings;
+        this.configPath = configPath;
+    }
+
+    /**
+     * This function wraps around all rest requests
+     * If the request is authenticated, then it goes through
+     */
+    public RestHandler wrap(RestHandler original) {
+        return new RestHandler() {
+
+            @Override
+            public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+                org.apache.logging.log4j.ThreadContext.clearAll();
+                if (checkAndAuthenticateRequest(request, channel, client)) {
+                    original.handleRequest(request, channel, client);
+                }
+            }
+        };
+    }
+
+    // True is authenticated, false if not - this is opposite of the Security plugin
+    private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
```

Comment:
This is a funky flow of control "if I return false don't execute more code".  I'd rather use exception handling and have it caught by an outer layer